src/Input.js

```diff
@@ -111,6 +111,7 @@ const Input = React.createClass({
         input = (
           <Button {...this.props} componentClass='input' ref='input' key='input' />
         );
+        delete this.props.bsStyle;
```

Comment:
We should not be mutating `this.props`. Future version of react will fail when doing this. They have plans to incorporate immutablejs with props.